pkg/apis/kubevirt/helper/scheme.go

```diff
@@ -123,3 +126,28 @@ func GetCloudProfileConfig(cluster *controller.Cluster) (*kubevirt.CloudProfileC
 
 	return &cloudProfileConfig, nil
 }
+
+// InfrastructureConfigFromInfrastructure extracts the InfrastructureConfig from the
+// ProviderConfig section of the given Infrastructure.
+func InfrastructureConfigFromInfrastructure(infra *extensionsv1alpha1.Infrastructure) (*api.InfrastructureConfig, error) {
+	config := &api.InfrastructureConfig{}
+	if infra.Spec.ProviderConfig != nil && infra.Spec.ProviderConfig.Raw != nil {
+		if _, _, err := decoder.Decode(infra.Spec.ProviderConfig.Raw, nil, config); err != nil {
+			return nil, err
+		}
+		return config, nil
+	}
+	return nil, fmt.Errorf("provider config is not set on the infrastructure resource")
+}
+
+// CloudProfileConfigFromCluster decodes the provider specific cloud profile configuration for a cluster
+func CloudProfileConfigFromCluster(cluster *controller.Cluster) (*api.CloudProfileConfig, error) {
```

Comment:
we already have a method for getting cloud profile config https://github.com/gardener/gardener-extension-provider-kubevirt/blob/master/pkg/apis/kubevirt/helper/scheme.go#L107

Reply:
consolidated